tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs

```diff
@@ -0,0 +1,253 @@
+                            asmProcessor.Process();
+
+                            lock (processedDirectories)
+                            {
+                                numAssemblies++;
+                                processedDirectories.Add(Path.GetDirectoryName(file) ?? string.Empty);
+                            }
+                        }
+                        catch (Datadog.Trace.Ci.Coverage.Exceptions.PdbNotFoundException)
+                        {
+                            // If the PDB file was not found, we skip the assembly without throwing error.
+                        }
+                        catch (BadImageFormatException)
+                        {
+                            // If the Assembly has not the correct format (eg. native dll / exe)
+                            // We skip processing the assembly.
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger?.Error(ex);
+                        }
+                    }
+                });
+
+            // Add Datadog.Trace dependency to the deps.json
+            if (processedDirectories.Count > 0)
+            {
+                foreach (var directory in processedDirectories)
+                {
+                    var version = typeof(Instrumentation).Assembly.GetName().Version?.ToString();
+                    foreach (var depsJsonPath in Directory.EnumerateFiles(directory, "*.deps.json", SearchOption.TopDirectoryOnly))
+                    {
+                        var json = JObject.Parse(File.ReadAllText(depsJsonPath));
+                        var libraries = (JObject)json["libraries"];
+                        libraries.Add($"Datadog.Trace/{version}", JObject.FromObject(new { type = "reference", serviceable = false, sha512 = string.Empty }));
+
+                        var targets = (JObject)json["targets"];
+                        foreach (var targetProperty in targets.Properties())
+                        {
+                            var target = (JObject)targetProperty.Value;
+
+                            target.Add(
+                                $"Datadog.Trace/{version}",
+                                new JObject(
+                                    new JProperty(
+                                        "runtime",
+                                        new JObject(
+                                            new JProperty(
+                                                "Datadog.Trace.dll",
+                                                new JObject(
+                                                    new JProperty("assemblyVersion", version),
+                                                    new JProperty("fileVersion", version)))))));
+                        }
+
+                        using (var stream = File.CreateText(depsJsonPath))
```

Comment:
This could throw (e.g. we don't have permission to write it), should we handle it/log?

Reply:
Handled in [3257cb2](https://github.com/DataDog/dd-trace-dotnet/pull/2739/commits/3257cb2a2db0bd5fed92b5b64cff1ea50d664468)